### PR TITLE
Implements Status Conditions For MachinePool

### PR DESCRIPTION
**Pull request by NicoleYson**

### CODE REVIEW

File: exp/controllers/awsmachinepool_controller.go [hunk 1]
```diff
@@ -145,7 +147,24 @@ func (r *AWSMachinePoolReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
...
 	}
 
-	// todo: defer conditions + machinePoolScope.Close()
+	// Always close the scope when exiting this function so we can persist any AWSMachine changes.
+	defer func() {
+		// set Ready condition before AWSMachine is patched
+		conditions.SetSummary(machinePoolScope.AWSMachinePool,
+			conditions.WithConditions(
+				infrav1.InstanceReadyCondition,
+				infrav1.SecurityGroupsReadyCondition,
```
Comment (rudoi) [inline exp/controllers/awsmachinepool_controller.go:156]: i'm not sure if either of these conditions apply to ASGs - we may want to define some new ones in expinfrav1

Comment (rudoi) [reply exp/controllers/awsmachinepool_controller.go:156]: unless InstanceReadyCondition is part of the contract

File: exp/controllers/awsmachinepool_controller.go [hunk 2]
```diff
@@ -224,6 +252,30 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(_ context.Context, machinePoo
 	// Set state
 	// Reconcile AWSMachinePool State
 	//Handle switch case instance.State{}
+	// switch instance.State {
```
Comment (rudoi) [inline exp/controllers/awsmachinepool_controller.go:255]: these comments needed? i have a feeling they may conflict with #31

Comment (NicoleYson) [reply exp/controllers/awsmachinepool_controller.go]: This was me not pushing 🤦‍♀️

File: exp/controllers/awsmachinepool_controller.go (hunk 1 above)
Comment (NicoleYson) [reply exp/controllers/awsmachinepool_controller.go:156]: I am fairly sure it was a const that was defined in that status struct. I can make it more relevant. Was unsure how much I was allowed to deviate. 
I will resolve this in the next commit.

**Pull request merged**